qt_binder/tests/test_widgets.py

```diff
@@ -56,8 +88,42 @@ def test_field_format_func(self):
 
     def test_field_text_edited(self):
         slider = RangeSlider()
-        self.assertEqual(slider.value, 0)
-        self.assertEqual(slider.slider.value, 0)
-        slider.field.trait_property_changed('textEdited', Undefined, 20)
-        self.assertEqual(slider.value, 20)
-        self.assertEqual(slider.slider.value, 20)
+        slider.construct()
+        try:
+            slider.configure()
+            self.assertEqual(slider.value, 0)
+            self.assertEqual(slider.slider.value, 0)
+            slider.field.text = '20'
+            slider.field.trait_property_changed(
+                'editingFinished', Undefined, True)
+            self.assertEqual(slider.value, 20)
+            self.assertEqual(slider.slider.value, 20)
+        finally:
+            slider.dispose()
+
+    def test_validator_states(self):
+        # When the validator is Invalid or Intermediate, the value of the
+        # RangeSlider should not change.
+        range_slider = RangeSlider(range=(30, 60), value=50)
+        range_slider.construct()
+        try:
+            range_slider.configure()
+            slider = range_slider.slider
+            field = range_slider.field
+
+            self.assertEqual(slider.value, 50)
+            self.assertEqual(range_slider.value, 50)
+
+            # State is Intermediate.
+            field.text = '1'
+            field.trait_property_changed('editingFinished', Undefined, True)
```

Comment:
Side question: is it worth wrapping up this pattern nicely somewhere? I assume that you did this to convince the `Binder` that the `editingFinished` event fired without actually triggering the Qt signal.